presidio-anonymizer/presidio_anonymizer/entities/engine/result/engine_result_item.py

```diff
@@ -0,0 +1,20 @@
+from abc import ABC, abstractmethod
+
+
+class EngineResultItem(ABC):
```

Comment:
We have quite a few classes holding the same thing. `EngineResultItem`, `AnonymizedEntity`, `RecognizerResult`, `TextMetadata` , `EncryptResult` ,`DecryptedEntity`. Consider merging into one class that everybody more or less uses or at least inherits from. IMHO we probably don't need more than two (RecognizerResult and another one focusing on the output)

Reply:
I think merging them into one will limit you to do changes in the future. 
If the anonymizer and decryptor will need to return something else and do different logics it will cripple the code.
I think in terms of design and looking into the future this is more robust and correct.
Also, you are trying to unify the results and the responses which will is a bad practice, the fact that entities looks the same doesn't mean they need to be unified. They have different purposes and logics. This about future expansions.

Reply:
Agree that sometimes unifying isn't the right thing to do. Are we sure we need all six?